Modified

horobito opened the pull request:
하...

horobito committed:
기존 오류 제거 및 friend 도메인 영역 추가

horobito committed:
기존 오류 제거 및 friend 도메인 영역 추가

horobito committed:
Merge branch 'modified' of https://github.com/mrfourfour/horobito_spring_server into modified

aerain commented on src/main/java/com/example/demo/friend/controller/FriendShipController.java:
```diff
@@ -0,0 +1,38 @@
+
+    @PostMapping("/{friendId}")
+    public String dealWithFriendShip(@PathVariable Long friendId){
+        String result = friendShipService.dealWithFriendShip(friendId);
```
메소드의 결과값은 해당 도메인에 대한 값 객체를 반환시키도록 해야 합니다.
String은 해당 도메인에 대한 값을 평가하기엔 너무 추상적이어서 알기 힘드네요.

aerain commented on src/main/java/com/example/demo/friend/domain/Friend.java:
```diff
@@ -1,37 +1,38 @@
 @Entity
-@Table(name = "friend")
-@Getter
 @Setter
```
`@Setter`가 외부에 열려있는 전략은 좋은 선택이 아닙니다.
누가 갑자기 님 팔 떼다가 강철의 연금술사마냥 무쇠 팔뚝 달아둔다고 생각하면 이해하기 편하겠네요.
적어도 해당 도메인 레이어에서만 수정할 수 있도록 Package-private정도로 접근제한자를 지정하면 좋을 것 같습니다.

aerain commented on src/main/java/com/example/demo/friend/domain/Friend.java:
```diff
@@ -1,37 +1,38 @@
-    public Friend(User friendUser){
-        this.frinedId = friendUser.getId();
-        this.friendname = friendUser.getUserId();
+    public Friend(FriendId friendId, FriendName friendname){
```
밑에 createFriend라는 정적 팩토리 메소드가 있는 것으로 보입니다.
정적 팩토리가 있을 경우에는 해당 생성자는 private등으로 외부에 노출 시키지 않는 전략을 취해야 합니다.

aerain commented on src/main/java/com/example/demo/friend/domain/FriendId.java:
```diff
@@ -0,0 +1,21 @@
+import javax.persistence.Embeddable;
+
+@Getter
+@Setter
```
여거도 마찬가집니다.

aerain commented on src/main/java/com/example/demo/friend/domain/FriendName.java:
```diff
@@ -0,0 +1,22 @@
+import javax.persistence.Embedded;
+
+@Getter
+@Setter
```
마찬가지

aerain commented on src/main/java/com/example/demo/friend/domain/Friendship.java:
```diff
@@ -0,0 +1,44 @@
+
+    private Boolean friendState;
+
+    public Friendship(User user, Friend friend){
```
요것도 private

aerain commented on src/main/java/com/example/demo/friend/controller/FriendShipController.java:
```diff
@@ -0,0 +1,38 @@
+        String result = friendShipService.dealWithFriendShip(friendId);
+
+        if (result.equals("Accept") || result.equals("Already accept")){
+            return "200 OK";
```
```suggestion
ResponseEntity.ok()
```
문자열 그만쓰세요...

horobito committed:
User 도메인의 Setter 접근자 변경

외부에 Setter가 열려있는 현상 방지

horobito committed:
Friend 애그리거트 내의 도메인들의 Setter 접근자 변경

외부에 Setter가 열려있는 현상 방지

horobito committed:
변경내역 : FriendId와 FriendName에 어노테이션 추가

이유
에러메세지
Error:(14, 14) Class 'FriendId' should have [public, protected] no-arg constructor
Error:(15, 14) Class 'FriendName' should have [public, protected] no-arg constructor

로 인한 생성자 자동생성 어노테이션 추가

horobito committed:
변경내역
Friend의 @Setter에 접근제한자 설정

이유
@Setter에 접근 제한을 걸어야 하는데, @Getter에 걸어서 변경함

horobito committed:
변경내역
생성자의 접근제한자 private로 설정

이유
정적 팩토리 메소드로 인해 그것이 사용하는 생성자의 접근제한자를
private로 설정하여 외부에 노출시키지 않게 하기 위해

horobito committed:
변경내역
생성자의 접근제한자 변경으로 인한 코드 오류 수정

이유
접근제한자 변경 및 기존 작성된 코드가 잘못 작성되어 수정함
-> 직접 생성자 호출 대신, 정적 팩토리 메소드를 사용하여 호출하도록 수정

horobito committed:
변경내역
FriendShipController에서
String 보내는 것을 ResponseEntity를 사용하도록 변경

이유
문자열 직접 보내는 것 말고, 응답에 설정하여 보내는 것이 맞기 때문

horobito committed:
변경내역
FriendShipService에서 사용할 Enum 클래스 생성

이유
문자열 보내는 대신, enum 사용하여 상태 분류

horobito committed:
변경내역
FriendShipService에서 Enum클래스 사용

이유
문자열 보내는 대신, enum 사용하여 상태 분류

horobito committed:
변경내역
FriendShipController 에서 Enum 클래스 사용하도록 변경

이유
문자열 보내는 대신, enum 사용하여 상태 분류

horobito committed:
변경내역
user의 이름과 비밀번호를 저장할 값 객체 생성

이유
getPassword.getPassword와 같은 가독성 저하를 막기 위해 대체재로 설정

horobito committed:
*변경내용
UserBasicInfo를 User에 적용

*이유
기존 Username 과 Password를 UserBasicInfo로 대체하기 위해

horobito committed:
*변경내용
기존 getUsername과 getPassword 사용 대신 UserBasicInfo를 사용하도록 적용

*이유
기존 Username 과 Password를 UserBasicInfo로 대체하기 위해

horobito committed:
*변경내용
password와 username을 UserBasicInfo 에 넝음으로써,
기존 User 내부의 Password와 Username을 제거

*이유
중복된 내용이므로 삭제

horobito committed:
*변경내역
- 생성자 직접 호출이 아닌 간접 방식으로 변경

*이유
- service 층에서 domain층을 직접 생성시키지 않게 하기 위해

horobito committed:
*변경내역
- 생성자 직접 호출이 아닌 간접 방식으로 변경(2)

*이유
- service 층에서 domain층을 직접 생성시키지 않게 하기 위해 (2)

horobito committed:
*변경내역
- 생성자 직접 호출이 아닌 간접 방식으로 변경(3)
- 공통부분 묶어 하나로 합침

*이유
- service 층에서 domain층을 직접 생성시키지 않게 하기 위해 (3)
- 중복 방지

horobito committed:
*변경내역
- FriendController.java 제거

*이유
- FriendShipController와 중복된 기능이므로 제거

horobito committed:
*변경내역
- 친구 삭제 구현

*이유
- 위와 동일

horobito committed:
*변경내역
- 친구 요청 확인 controller 층 구현

*이유
- 친구 요청 확인 기능 생성하기 위함

horobito committed:
*변경내역
- 친구 요청 확인 구현

*이유
- 친구 요청 확인 기능 생성

horobito committed:
*변경내역
- UserRepository의 findByUsername을
findByUserBasicInfo_Username으로 변경

*이유
- 수정한 User 도메인에 맞춰 변경

horobito committed:
*변경내역
- 오타 수정 및 기존 변경내용 미적용된 부분들 적용

*이유
- 변경된 내용으로 적용이 안되서 컴파일 오류 발생

horobito committed:
*변경내역
- 오타 수정

*이유
- 오타

horobito committed:
*변경내역
- 오타 수정

*이유
- 오타

horobito committed:
*변경내역
- 중첩된 도메인 존재
  [Feed와 Post]
  로 인해 Post 도메인 제거 및 실수로 지운 Friend 도메인 다시 살림...

*이유
- 중첩 및 실수 백업

horobito committed:
*변경내역
- Feed의 content 타입을 변경하였으므로, 변경된 내용으로 반영

*이유
- 위와 동일

horobito committed:
*변경내역
- @Transaction 변경 및 username에 column값 설정

*이유
- 서버 작동은 되지만, 회원가입 시 DB에 저장이 안되는 이유 찾는 과정

aerain commented on src/main/java/com/example/demo/feed/domain/Content.java:
```diff
@@ -0,0 +1,27 @@
+import javax.persistence.Embeddable;
+
+@Getter
+@Setter
```
항상 말하던거

aerain commented on src/main/java/com/example/demo/feed/domain/Content.java:
```diff
@@ -0,0 +1,27 @@
+@Getter
+@Setter
+@Embeddable
+@NoArgsConstructor
```
파라미터가 없는 생성자가 비즈니스에 필요한 게 아니라면 Protected로 제한자를 좁히는게 좋겠네요

aerain commented on src/main/java/com/example/demo/feed/domain/Feed.java:
```diff
@@ -24,7 +25,8 @@
+    @Embedded
+    private Content content;
 
     private Long likeNum;
```
Num이라는 명칭으로 이름에 등록하는거는 좋은 방법이 아닙니다. 자료형으로 표현해야합니다.

aerain commented on src/main/java/com/example/demo/feed/domain/Feed.java:
```diff
@@ -47,7 +49,7 @@ public void delete(){
 
 
-    public static Feed createFeed(Writer writer, String contents) {
+    public static Feed createFeed(Writer writer, Content contents) {
```
contents? content?

aerain commented on src/main/java/com/example/demo/feed/service/CommentService.java:
```diff
@@ -28,7 +29,8 @@
         Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
         Feed feed = feedRepository.findFeedByIdAndIsDeleted(feedId, false);
-        User user = userRepository.findByUserId(authentication.getName());
+        Username username = Username.createUsername(authentication.getName());
```
이미 Username의 클래스의 팩토리 메소드이기 때문에 굳이 create**Username**이라는 suffix를 둘 필요는 없어보입니다.

aerain commented on src/main/java/com/example/demo/feed/service/FeedService.java:
```diff
@@ -51,10 +45,22 @@ public Feed findFeedDetailByFeedId(Long id) {
         feedRepository.save(feed);
     }
+
+    public User findUserByAuthentication(){
```
By? From?

Authentication 을 파라미터로 받을것처럼 생겼는데 현재 메소드명으로는 그런게 잘 표현되어있지 않네요.

aerain commented on src/main/java/com/example/demo/friend/controller/FriendShipController.java:
```diff
@@ -0,0 +1,52 @@
+
+    @GetMapping
+    // request query로 받는 값들
+    public List<FriendDto> getMyFriendList(@RequestParam(value="page") int page,
```
`getMyFriendList`

List 등의 자료구조에 대한 내용을 직접 표현하지 말아주세요. 비즈니스 특성에 따라 자료구조는 언제든지 바뀔 수 있습니다. 자료구조의 변경이 메소드명의 변경으로 전파되지 않게 해야합니다.

aerain commented on src/main/java/com/example/demo/friend/controller/FriendShipController.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    @DeleteMapping("/{friendId}")
+    public void deleteFriendShipRequest(@PathVariable Long friendId){
```
`deleteFriendShip` 정도가 좋겠네요

horobito replied:
여기까지 개선항목들 코드에 적용함

aerain commented on src/main/java/com/example/demo/friend/domain/FriendShipRepository.java:
```diff
@@ -0,0 +1,18 @@
+
+public interface FriendShipRepository  extends JpaRepository<Friendship, Long> {
+
+    List<Friendship> findAllByUser(User user, Pageable pageable);
```
Pageable을 사용할거면 반환형은 `Page<Friendship>` 이 되어야 할 것 같습니다.

horobito replied:
여기까지 개선항목들 코드에 적용함

aerain commented on src/main/java/com/example/demo/friend/service/FriendDto.java:
```diff
@@ -0,0 +1,16 @@
+
+import lombok.Getter;
+import lombok.Value;
+@Getter
```
`@Value` 어노테이션을 사용할때는 `@Getter` 는 생략 가능합니다.

aerain commented on src/main/java/com/example/demo/friend/service/FriendShipService.java:
```diff
@@ -0,0 +1,117 @@
+    public List<FriendDto> getMyFriendList(int page, int size) {
+        Username username = Username.createUsername("jihwan");
+        User user = userRepository.findByUserBasicInfo_Username(username);
+        List<FriendDto> friendshipList = friendShipRepository.findAllByUser(user, PageRequest.of(page, size))
+                .stream()
+                .filter(Friendship::getFriendState)
+                .map(Friendship::getFriend)
+                .map(this::toFriendDto)
+                .collect(Collectors.toList());
```
올 스트림 👍

aerain commented on src/main/java/com/example/demo/friend/service/FriendShipService.java:
```diff
@@ -0,0 +1,117 @@
+
+
+    @Transactional
+    public Result dealWithFriendShip(Long friendId) {
```
친구관계를 어떻게 처리하는걸까요? 친구를 추가한다? 친구를 삭제한다?
establish 나, 오히려 간단하게 create정도로 정해도 좋을것 같네요.

메소드명만 보고서는 이게 뭐하는 넘인지 잘 모를것 같습니다.

aerain replied:
Result는 이 도메인을 표현하기엔 너무 추상적인거 같네요
`FriendShipResult`나 아니면 좀더 이 친구 도메인을 잘 표현할 만한 결과정도로 표현하면 좋을 것 같습니다.

aerain commented on src/main/java/com/example/demo/user/domain/UserBasicInfo.java:
```diff
@@ -0,0 +1,33 @@
+        this.password = password;
+    }
+
+    public String getPasswordInfo(){
```
Info suffix를 굳이 붙여둔 이유가 있나요?

horobito committed:
*변경내역
- 1)기존 Post 도메인을 Feed로 변경하면서
  놓친 @Setter와 @NoArgcCons... 접근제한자 설정 및 다른 부분도 확인 후 설정
- 2) application.properties 에 중첩된 의존성 삭제

* 이유
- 기존 Post에 했던걸 변경할 때 못보고 지나침 및 확인하지 못한 부분들 있었음
- 의존성 넣을 때 잘못 넣음 - 주의할 것

horobito committed:
*변경내역
- 1)기존 isDeleted 를 deleted로 변경

* 이유
- 다른 사람이 읽을 때 가독성 저하 및 문법오류 등 문제 발생

horobito committed:
*변경내역
- 1)기존 likeNum 를 preference 로 변경

* 이유
- Num이란 명칭으로 등록하는 방법은 좋지 않음

horobito committed:
*변경내역
- 1)도메인 영역 contents , content 를 content로 통일
- 2) service 층에서 외부 입력값인 contents를
     insertedContent로 변경

* 이유
- 헷갈림 방지 및 무엇인지 명확히 하기 위해

horobito committed:
*변경내역
- 1) 클래스의 팩토리 메소드들에 붙어있는 불필요한 접미사 제거
  ex) createUsername -> create

* 이유
- 불필요한 접미사 제거

horobito committed:
*변경내역
- 1) 메소드 이름을 좀 더 알맞게 표현

* 이유
- 이름으로 혼동될 수 있는 것 방지

horobito committed:
*변경내역
- 1) 메소드 이름에 자료구조 내용 나와있는 것 제거

* 이유
- 자료구조는 언제든지 변경될 수 있으므로,
  메소드 이름에 자료구조가 나와있는 건 추후 문제가 될 가능성 있음

horobito committed:
*변경내역
- 1) 메소드 이름에 불필요한 부분 제거
    deleteFriendShipRequest -> deleteFriendship

* 이유
- 좀 더 간단하고 알기 쉽게 작성하기 위함

horobito committed:
*변경내역
- 1)  반환형을 page로 변경

* 이유
- Pageable를 사용하니 반환형도 그것에 알맞게 변경

horobito committed:
*변경내역
- 중첩되는 메소드 제거

*이유
- 필요없는 메소드 존재

horobito committed:
*변경내역
- 클래스명 변경(Result-> FriendShipResult)
- 메소드명 변경(dealWithFriendShip-> create)

*이유
- 도메인이 의미하는 바를 좀 더 명확하게
- 메소드명이 의미하는 바를 좀 더 명확하게

horobito committed:
*변경내역
- 메소드명 변경(getPasswordInfo-> getPassword)

*이유
- 필요없는 접미사 제거

horobito committed:
*변경내역
- 자료형으로 타입 변경

*이유
- 의미를 좀 더 명확히 나타냄 & 1개의 타입 2번 이상 사용하는 것 방지

horobito committed:
*변경내역
- security 도메인 생성

*이유
- 보안 관련 도메인 처리를 위해 생성

horobito committed:
*변경내역
- WebSecurityConfig 생성

*이유
- security 관련 설정에 필요한 클래스 생성
- 아직은 보안 관련 적용 하지 않을 것이라서
  모든 요청 허용으로 임시 설정

horobito committed:
*변경내역
- repository 내부 메소드 이름 변경

*이유
- Feed 내 변경된 필드명과 대응되도록 변경

horobito committed:
*변경내역
- WebSecurityConfig 내 필터 임시 제거

*이유
- 아직 불필요

horobito committed:
*변경내역
- 오타 수정

*이유
- 오타 수정

horobito committed:
*변경내역
- DB 칼럼 추가 및 이것에 맞게 매칭되는 column의 이름 변경

*이유
- DB 수정으로 인해

horobito committed:
*변경내역
- 테스트 위한 코드 임시 변경

*이유
- 위와 동일

WonBeomChoi commented on src/main/java/com/example/demo/user/application/LoginRequest.java:
```diff
@@ -0,0 +1,15 @@
+package com.example.demo.user.application;
```
이거는 왜 application으로 해놨나요 signupRequest는 service에 있는데....
저 requestmapping어케하는지 보려고왔다가 레포지토리 구성 따라하려고했는데 망했네요 제 시간 물어내세요

WonBeomChoi commented on src/main/java/com/example/demo/user/service/SignupRequest.java:
```diff
@@ -0,0 +1,13 @@
+package com.example.demo.user.service;
```
이거는 왜 service로 해놨나요 loginRequest는 application에 있는데....
저 requestmapping어케하는지 보려고왔다가 레포지토리 구성 따라하려고했는데 망했네요 제 시간 물어내세요..

WonBeomChoi reviewed (changes requested):
통일 부탁드립니다 지금 우리나라 분단국가라고 로그인하고 회원가입 분단시켜놨는데 이게 민족정서에 맞는 코딩입니까?
현재 한국 정서에 맞는 개발 부탁드립니다.

aerain commented:
이거 각 객체들 단위테스트 짜보면 좋을거같네요 JUnit으로 작성해보시죠

horobito committed:
* 변경내역
- LoginRequest 의 위치 변경

*이유
- 한 곳에 통일시키기 위함

horobito committed:
* 변경내역
- Content 도메인 내 필드 타입 변경 및 그에 의존하는 코드 변경

*이유
- 값 객체 사용하여 표현하기 위해

horobito committed:
* 변경내역
- Preference 에 초기값 부여

*이유
- null 값 없애기 위해

horobito committed:
* 변경내역
- Feed 생성자에 생성 시간 초기화 작성

*이유
- 초기화 미실시로 인한 오류 해결

horobito committed:
* 변경내역
- Preference 원래대로 수정

*이유
- 기존 코드에 초기값을 설정해도 DB에 값이 안들어가서 굳이 바꿀 필요가 없음

horobito committed:
* 변경내역
- Preference 정적 메소드 통한 생성자 호출하도록 조정
- 기존 Preference에 초기값 넣어주는 것 다시 원래대로 조정
- 위 정적 메소드를 Feed 에서 사용하도록 조정

*이유
- 기존 Feed 생성 시 사용하도록
- 굳이 할 필요가 없어서
- 초기화 안해서 null 값 들어가는 것 방지

horobito committed:
* 변경내역
- 생성자 접근 제한자 조정

*이유
-패키지 내부에서만 다룰 수 있도록 하기 위해

horobito committed:
* 변경내역
- 종아요 누르는 코드 대략적인 모습 생성
*이유
- 좋아요 기능 구현 위해

horobito committed:
* 변경내역
- 매개변수 이름 조정
*이유
- feed 내 필드명에 맞게 하려고

horobito committed:
* 변경내역
- Feed 내 deleted 값 생성 시 초기화하도록 조정
- FeedRepository 내 매개변수 타입 조정
-
*이유
- null 이 아닌 false 가 들어가도록 조정
- feed 내 필드명에과 일치하도록

horobito committed:
* 변경내역
- FeedService 에 좋아요 기능 구현
-
*이유
- 위와 동일

horobito committed:
Merge branch 'modified' of https://github.com/mrfourfour/horobito_spring_server into modified

horobito committed:
* 변경내역
- Preference 에 초기값 넣어줌

*이유
- 없으면 null값 들어가서 [이전 수정 내용에는 잘못 이해한 내용 반영되었음]

horobito committed:
* 변경내역
- Controller 에서 발견된 오타 수정
 1) Controller 이름 잘못된 것 수정
 2) path variable 로 받을 때 그 http 요청에 적혀있는 이름대로 안되어있는 것
   바로 잡음

*이유
- 잘못된 코드로 인한 오류 원인 제거

horobito committed:
* 변경내역
- Comment 의 도메인 및 서비스 수정

*이유
- 1) 테스트를 위한 service 층 코드 수정
- 2) comment 도메인 내 preference 변경 내용 수정

horobito committed:
* 변경내역
- Comment 에서 받는 형식 수정

*이유
- json 형식으로 보내기 위해 수정

horobito committed:
* 변경내역
- Comment 도메인의 작성 시간 초기화 코드 추가

*이유
- 초기화 코드가 없어서 추가함

horobito committed:
* 변경내역
- CommentService 에서 직접 get 하는 것 막기

*이유
- 머리 말고 팔한테 직접 명령하는 것과 같은 행위 방지

horobito committed:
* 변경내역
- service층에서 문자열로 응답 보내는 동작 삭제

*이유
- 문자열 다루는 것 지양

horobito committed:
* 변경내역
- 테스트 위한 user 생성코드 임시로 변경

*이유
- 친구 기능 테스트

horobito committed:
* 변경내역
- Path 오타 변경

*이유
- 오타

horobito committed:
* repository 매개변수 타입 변경(long->FriendId) 으로 인한 그에 의존하는 모든 클래스들을 수정

horobito committed:
* 단순한 줄나눔

horobito committed:
* User 를 대체할 새로운 도메인 생성

horobito committed:
* User 를 대체할 새로운 도메인 생성(2)

horobito committed:
* User 를 대체할 새로운 도메인 UserInfo 완성

horobito committed:
* User 에 있는 FriendShip 을 UserInfo 로 이식

horobito committed:
* User 를 UserInfo로 수정

horobito committed:
* Friendship 에서 User 를 UserInfo 로 바꿔서
  이것에 의존성을 가진 것들도 수정된 것 (
  -> 잘못 바뀐 것도 있어서 수정
  -> 현재 대부분 수정 완료 했고,
     FriendService 부분 수정중

aerain commented on src/main/java/com/example/demo/feed/domain/Feed.java:
```diff
@@ -24,35 +25,50 @@
 
-    @Column(name = "is_delete")
-    private Boolean isDeleted;
+    @Column(name = "deleted")
```
동일 컬럼명의 경우 어노테이션 생략 가능합니다.

aerain commented on src/main/java/com/example/demo/feed/domain/Feed.java:
```diff
@@ -24,35 +25,50 @@
 
-    private Feed(Writer writer, String content){
+    @OneToMany(mappedBy = "feed", cascade = {CascadeType.PERSIST})
+    @JsonIgnoreProperties("feed")
```
표현층의 책임을 도메인 계층에 짊어지게하지 마세요

aerain commented on src/main/java/com/example/demo/feed/domain/Writer.java:
```diff
@@ -24,7 +25,7 @@ public Writer(Long id, String wrtName) {
 
-    public static Writer makeWriter(User user) {
-        return new Writer(user.getId(), user.getUserId());
+    public static Writer create(User user) {
```
서로 다른 도메인 계층이 직접 서로 의존하지 않도록 하세요.

aerain replied:
도메인 간 의존하는 대신 응용층에서 둘 사이의 결합을 하는게 좋겠네요

aerain commented on src/main/java/com/example/demo/feed/service/CommentService.java:
```diff
@@ -25,27 +19,33 @@
+//        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        Feed feed = feedRepository.findFeedByIdAndDeleted(feedId, false);
+//        Username username = Username.create(authentication.getName());
+        Username username = Username.create("jihwan");
```
하드코딩..?

aerain replied:
테스트코드로 작성하세요

aerain commented on src/main/java/com/example/demo/friend/service/FriendShipService.java:
```diff
@@ -0,0 +1,127 @@
+
+    @Transactional
+    public FriendShipResult create(Long inputId) {
+        UserName username = UserName.create("jihwan");
```
여기도..

horobito committed:
*
1) FriendShipService 에서 필요없는 의존성 제거

2) FriendShipController 에서
  친구관계 처리 메소드 이름을 좀 더 명확하게 수정

horobito committed:
* 테스트용 하드코딩 제거

horobito committed:
* 동일 컬럼명이라서 어노테이션 생략

horobito committed:
* Writer 필드 변경 및  그것에 의존하는 것들 변경
* 하나씩 확인해가며 하드코딩한 부분 삭제 중

aerain commented on src/main/java/com/example/demo/feed/domain/Preference.java:
```diff
@@ -0,0 +1,32 @@
+    private Long preference = 0L;
+
+    public void like(){
+        this.preference ++;
```
내부적으로 필드를 ++ 한다면 preference를 사용하는 다른 스레드에서 데이터 불일치가 생길 수 있습니다.
내부 값 변경보다 새로운 값 객체를 생성하도록 하는게 좋을것 같습니다.

```java
public Preference like() {
    return new Preference(this.preference + 1);
}
```

aerain commented on src/main/java/com/example/demo/feed/domain/Preference.java:
```diff
@@ -0,0 +1,32 @@
+public class Preference {
+
+    @Column(name = "preference")
+    private Long preference = 0L;
```
preference가 not null 함이 보장되면 Long 타입보단 long 타입을 사용하는게 좋겠군요

horobito committed:
* Preference의 타입 변경 & 내부적 값 변경 대신 새로운 객체 생성으로 대체

horobito committed:
* Writer에 사용될 값 객체 WriterId 와 WriterName 생성

horobito committed:
* 이름을 좀 더 명확하게 변경

horobito committed:
* FriendId, FriendName, UserId, UserName 으로 중복되어 있는 것들
  모두 Identification, Name 으로 통일

horobito committed:
* Identification, Name 적용하여 FriendShipService 수정 시작

horobito committed:
* 하드코딩 하지 않게 변경

horobito committed:
* 하드코딩 하지 않게 변경

horobito committed:
* 변경된 사항들 적용하도록 수정

horobito committed:
* 요구사항 반영하여 수정

horobito committed:
* 열거상수 추가 및 세부적인 요구사항에 맞게 수정

horobito committed:
* 변경된 도메인에 맞게 수정

horobito committed:
* 잘못된 코드 수정 및 DTO 중복 피하기 위한 코드 수정

horobito committed:
* 필요없는 코드 제거

horobito committed:
* 메소드 추출

horobito committed:
* 메소드 추출 적용 및 필요없는 코드 제거

horobito committed:
* 타임라인 구현중
 - repository에서 친구 id 목록 가지고 어떻게 해야할 지 고민중

horobito committed:
* 타임라인 구현중
 - service 에서 어떻게 해야 Pageable 이 될 지 고민중

horobito committed:
* 타임라인 구현중
 - Pageable 구현 중 Sort.sort 어떻게 해보면 될 것 같은 느낌

horobito committed:
* 테스트 과정 중 필요없는 부분 및 잘못 짠 코드들  일단 주석 처리

aerain commented:
타임라인 구현은.. 다른 pr에서 ㅠㅠ

aerain commented:
테스트코드 먼저 짜라고했는데 왜 자꾸 구현만 하는겨...

horobito committed:
* FriendShipService 테스트 1 - FriendShip 생성 테스트

horobito committed:
* FriendShipService 테스트 1 - FriendShip 생성 테스트

aerain commented on src/main/java/com/example/demo/friend/service/FriendShipService.java:
```diff
@@ -86,6 +89,17 @@ public FriendShipResult create(Long inputedFriendId) {
+
+    }
+
+    public Authentication findAuthentication(){
+        return SecurityContextHolder.getContext().getAuthentication();
+    }
```
요런거는 책임을 다른 서비스에게 위임하면 어떨까요?
FriendshipService에서 해당 서비스를 모킹해서 원하는 Authentication을 반환하도록 stubbing 할 수 있습니다.
